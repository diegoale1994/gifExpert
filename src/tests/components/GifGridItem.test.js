const { shallow } = require("enzyme")
const { GifGridItem } = require("../../components/GifGridItem")

describe('pruebas a GridItem', () => {

    const title = 'Gif Example';
    const url = 'https:/localhost:8080/test.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test('should mostrar el componente', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de tener un parrafo con el titulo ', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })

    test('debe de tener la imagen = al prop y alt del titulo', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })

    test('que el div tenga la clase css', () => {
        const div = wrapper.find('div');
        expect(div.prop('className').includes('animate__fadeIn')).toBe(true)
    })

})
