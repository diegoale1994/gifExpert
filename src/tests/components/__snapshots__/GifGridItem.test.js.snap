// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas a GridItem should mostrar el componente 1`] = `
<div
  className="card animate__fadeIn"
>
  <img
    alt="Gif Example"
    src="https:/localhost:8080/test.jpg"
  />
  <p>
    Gif Example
  </p>
</div>
`;
